torch/_subclasses/meta_utils.py

```diff
@@ -242,7 +242,10 @@ def sym_sizes_strides_storage_offset(t):
                     # TODO: Delete this assert, and just attempt making the
                     # sparse tensor anyway; even if there is a shape_env, this
                     # tensor might be all static
-                    assert shape_env is None, "symbolic on sparse NYI"
+                    import sympy
+
+                    static = all(isinstance(s, (int, sympy.Integer)) for s in t.size())
```

Comment:
You should never have a sympy.Integer out of size, it's impossible

Reply:
We assert that, right?

Reply:
Oh, understood, misread, will update.